### DISCUSSION TITLE
EGUI Speed handle data records

### DISCUSSION THREAD

**Discussion by mjehrhart** [emilk/egui#1169] [Q&A]
I am wondering if there has been any apps that use  egui to handle around 150,000 records.  If I built my own widget to emulate a table, could egui handle this amount.  I have a terminal app which I am converting into a gui.

I appreciate any help on this topic. Thanks.

**Comment by mjehrhart** [comment]
So I am going to give the first answer to my own question.  I created a simple gui which loads about 152,000 records as buttons as once from a terminal app.  I implemented a scrollbar after some research.  And I am happy to say that there's no issue with speed or lagtime.  

Great stuff.

**Reply by emilk** [reply]
Wow, that's great! If performance becomes an issue you can use `ScrollArea::show_rows` to only paint the buttons that are actually visible.

**Reply by mjehrhart** [reply]
Thanks for the info.  So far so good.  I/m pulling about 150k files at a time.
